File: Web.Admin/ClientApp/src/app/models/AcvpUser/AcvpUser.ts
import { Person } from '../person/Person';

export class AcvpUser {
  certificateBase64: string;
  commonName: string;
  seed: string;
  expiresOn: Date;
  issuer: string;
  acvpUserID: number;
  personID: number;
  person: Person;
  fullName: string;
  companyId: number;
  companyName: string;
}

File: Libraries/Shared/NIST.CVP.Libraries.Shared.ACVPCore.Abstractions/Models/AcvpUser.cs
using System;
using System.Security.Cryptography.X509Certificates;

namespace NIST.CVP.Libraries.Shared.ACVPCore.Abstractions.Models
{
    public class AcvpUser : AcvpUserLite
    {
        private string _certificateBase64;
        public string CertificateBase64
        {
            get => _certificateBase64;
            set
            {
                _certificateBase64 = value;
                var cert = new X509Certificate(Convert.FromBase64String(_certificateBase64));
                
                ExpiresOn = DateTime.UnixEpoch;
                if (DateTime.TryParse(cert.GetExpirationDateString(), out var expiresOn))
                {
                    ExpiresOn = expiresOn;
                }

                Issuer = cert.Issuer;
            }
        }
        public string CommonName { get; set; }
        public string Seed { get; set; }
        public string Issuer { get; private set; }
    }
}